crowbar_framework/app/models/cinder_service.rb

```diff
@@ -205,6 +205,20 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # the VIP of the cluster to be setup
     allocate_virtual_ips_for_any_cluster_in_networks(controller_elements, vip_networks)
 
+    # Generate secrets uuid for libvirt rbd backend
+    proposal = ProposalObject.find_data_bag_item "crowbar/bc-cinder-default"
```

Comment:
> I'm mixed on automatically editing the proposal, especially as it will show the proposal as not applied in its latest version.

If this is the only concern, just pass `applied: true` as an option when saving the proposal.

Otherwise, I think this is a good solution. Doing this in the UI means we have to add UI code, then make sure that the API works the same way (auto-adds or at least requires the UUID field, then does not change it on subsequent updates). This sounds like extra work to me.